hal/common/retarget.cpp

```diff
@@ -529,10 +536,14 @@ extern "C" WEAK void mbed_sdk_init(void) {
 extern "C" int $Super$$main(void);
 
 extern "C" int $Sub$$main(void) {
-    mbed_sdk_init();
```

Comment:
mbed_sdk_init for ARMCC is moved to before the RTOS is initialized, but is still called in GCC after the RTOS has been initialized.  This behavior should be unified across the toolchains.